public/app/features/alerting/unified/hooks/useIsRuleEditable.ts

```diff
@@ -29,24 +27,25 @@ export function useIsRuleEditable(rulesSourceName: string, rule?: RulerRuleDTO):
 
   // Grafana rules can be edited if user can edit the folder they're in
   // When RBAC is disabled access to a folder is the only requirement for managing rules
+  // When RBAC is enabled the appropriate alerting permissions need to be met
   if (isGrafanaRulerRule(rule)) {
     if (!folderUID) {
       throw new Error(
         `Rule ${rule.grafana_alert.title} does not have a folder uid, cannot determine if it is editable.`
       );
     }
     return {
-      isEditable: hasEditPermission && folder?.canSave,
-      isRemovable: hasRemovePermission && folder?.canSave,
+      isEditable: contextSrv.hasAccess(rulePermission.update, folder?.canSave ?? false),
+      isRemovable: contextSrv.hasAccess(rulePermission.delete, folder?.canSave ?? false),
```

Comment:
nit: Can we move this back to separate variables like `hasEditPermission` and `hasRemovePermission`? It always makes it a bit easier to read the intentions and business logic of the code.